3rdparty/SConscript

```diff
@@ -319,6 +319,50 @@ elif 'sox' in system_dependencies:
 
     env = conf.Finish()
 
+# dep: sndfile
+if 'sndfile' in autobuild_dependencies:
+    sndfile_deps = []
+
+    if 'alsa' in autobuild_dependencies:
+        sndfile_deps += ['alsa']
+
+    if 'pulseaudio' in autobuild_dependencies:
+        sndfile_deps += ['pulseaudio']
+
+    env.BuildThirdParty(thirdparty_versions, 'sndfile', deps=sndfile_deps)
+
+    conf = Configure(env, custom_tests=env.CustomTests)
+
+    if not 'alsa' in autobuild_dependencies:
+        for lib in [
+                'asound',
+                ]:
+            conf.CheckLib(lib)
+
+    if meta.platform in ['darwin']:
+        env.Append(LINKFLAGS=[
+            '-Wl,-framework,CoreAudio'
+        ])
+
+    env = conf.Finish()
+
+elif 'sndfile' in system_dependencies:
+    conf = Configure(env, custom_tests=env.CustomTests)
+
+    if not conf.AddPkgConfigDependency('sndfile', '--cflags --libs', exclude_from_pc=True):
+        conf.env.AddManualDependency(libs=['sndfile'], exclude_from_pc=True)
+
+    if not is_crosscompiling:
+        if not conf.CheckLibWithHeaderExt(
+                'sndfile', 'sndfile.h', 'C',
+                expr='SNDFILE_LIB_VERSION_CODE >= SNDFILE_LIB_VERSION(1, 0, 28'):
```

Comment:
Could you clarify why 1.0.28 is minimal version that we support?

Ideally we should be able to use the oldest version among our CI images. Quick googling suggests that ubuntu 14.04 used libsndfile 1.0.25, and this is probably the oldest one, but ideally we should check other distros too.

See also https://roc-streaming.org/toolkit/docs/portability/supported_platforms.html

Reply:
@gavv the correct way to get libsndfile's version number is sf_version_string(). That was a copy and paste error on my part, sorry! 

However, I am struggling with how I can compare two string versions against each other in a similar fashion to SOX or libuv in a neat single line, since there is no macro that returns the version number as something immediately comparable. I know several long multi-line methods of parsing and converting a version strings to individual numbers for comparison in C, but I'm unsure if that would be okay or where/how to place that in the script.

Also you are correct, after researching all the available platforms version 1.0.25 is the oldest we must support.

Reply:
I think let's just skip version check for simplicity, i.e. omit `expr=`.